app/src/main/java/com/sun/premierleague/ui/team/TeamActivity.kt

```diff
@@ -0,0 +1,127 @@
+import com.sun.premierleague.data.model.Player
+import com.sun.premierleague.data.model.TeamItem
+import com.sun.premierleague.utils.RepositoryFactory
+import com.sun.premierleague.utils.loadImage
+import com.sun.premierleague.utils.showToast
+import kotlinx.android.synthetic.main.activity_detail_team.*
+
+class TeamActivity : BaseActivity(), TeamContact.View {
+
+    private var presenter: TeamPresenter? = null
+    private var adapterGoalkeeper = TeamAdapter()
+    private var adapterDefender = TeamAdapter()
+    private var adapterMidfielder = TeamAdapter()
+    private var adapterForwarder = TeamAdapter()
+
+    override val layoutResource: Int get() = R.layout.activity_detail_team
+
+    override fun initComponents() {
+        initAdapter()
+        initPresenter()
+        initCollapsingToolbar()
+        presenter?.idTeam = intent?.getStringExtra(ID_TEAM)
+        presenter?.start()
+    }
+
+    private fun initAdapter() {
+        recyclerGoalkeeper.adapter = adapterGoalkeeper
+        recyclerPlayerDefend.adapter = adapterDefender
+        recyclerPlayerMidfield.adapter = adapterMidfielder
+        recyclerPlayerForward.adapter = adapterForwarder
+    }
+
+    private fun initPresenter() {
+        val repository = RepositoryFactory.getRepository(this)
+        presenter = TeamPresenter(this, repository)
+    }
+
+    private fun initCollapsingToolbar() {
+        val collapsingToolbarLayout =
+            collapsingToolbarTeam as CollapsingToolbarLayout
+        val appBarLayout = appBarTeam as AppBarLayout
+        var isShow = true
+        var scrollRange = -1
+        appBarLayout.addOnOffsetChangedListener(AppBarLayout.OnOffsetChangedListener { barLayout, verticalOffset ->
+            if (scrollRange == -1) {
+                scrollRange = barLayout?.totalScrollRange!!
+            }
+            if (scrollRange + verticalOffset == 0) {
+                collapsingToolbarLayout.title = textDetailTeam.text
+                isShow = true
+            } else if (isShow) {
+                collapsingToolbarLayout.title = " "
+                isShow = false
+            }
+        })
+    }
+
+    override fun showTeam(data: List<TeamItem>) {
+        textDetailTeam.text = data[0].teamName
+        imageDetailTeam.loadImage(data[0].teamBadge)
+    }
+
+    override fun showCoach(data: List<Coache>) {
+        textNameCoachOne.text = data[0].coachName
+        textInfoCoachOne.text = String.format(
+            getString(R.string.text_info_coach_format),
+            data[0].coachAge,
+            data[0].coachCountry
+        )
+        if (data.size > 1) {
+            textNameCoachTwo.visibility = View.VISIBLE
+            textInfoCoachTwo.visibility = View.VISIBLE
+            textNameCoachTwo.text = data[1].coachName
+            textInfoCoachTwo.text = String.format(
+                getString(R.string.text_info_coach_format),
+                data[1].coachAge,
+                data[1].coachCountry
+            )
```

Comment:
Đoạn này dùng nhiều data[1] thì đặt thêm biến phụ cho đoạn code dễ hiểu hơn 
```suggestion
        if (data.size > 1) {
            val coach = data[1]
            textNameCoachTwo.visibility = View.VISIBLE
            textInfoCoachTwo.visibility = View.VISIBLE
            textNameCoachTwo.text = coach.coachName
            textInfoCoachTwo.text = 
                getString(R.string.text_info_coach_format, coach.coachAge, coach.coachCountry)
```